mmv1/templates/terraform/examples/cloudfunctions2_basic.tf.erb

```diff
@@ -1,7 +1,15 @@
 # [START functions_v2_basic]
+locals {
+  project = "ProjectId" # Google Cloud Platform Project ID
+}
+
+provider "google-beta" {
+   project = local.project
+}
+
 resource "google_storage_bucket" "bucket" {
   provider = google-beta
-  name     = "<%= ctx[:vars]['bucket_name'] %>"  # Every bucket name must be globally unique
+  name     = "${local.project}-<%= ctx[:vars]['bucket_name'] %>"  # Every bucket name must be globally unique
```

Comment:
Currently `local.project` is going to be `ProjectId` as it doesn't get changed. I'm assuming the attempt here is to add the project id to the bucket name? I think we want to avoid that as the bucket name has a limit of 63 characters and we're not sure how long the id of the project will be. The bucket name will already be globally unique as the way we're handling these variables, when running tests, we'll generate a random 10-character string to the end of it. Additionally we add `tf-test` to the front of them so when we're sweeping our test projects, we know which buckets are safe to delete. Does that make sense?

Reply:
Unfortunately, without prepending with provider/project the sample will not work for all users. The sample will also not be a good fit to be used in the official docs. We need for these samples to be approachable for new Terraform users as well. Besides turning off the tests, is there any way to inject the test project id from the terraform.yaml file?

Reply:
Sure, ok, so if you want to access the project name, we usually do it like [this in terraform.yaml](https://github.com/GoogleCloudPlatform/magic-modules/blob/5268d03a79ba5644c9b8252175f642245881c8b7/mmv1/products/dlp/terraform.yaml#L28)
then refer to it as `<%= ctx[:test_env_vars]['project'] %>`  in the example.
I see that project names have a length limit of 30 characters, so i think if we change the `bucket_name` var to something 10 characters or less (if my math is right) we can maybe make this `<%= ctx[:vars]['bucket_name'] %>-<%= ctx[:test_env_vars]['project'] %>`

which would in turn be `tf-test-my-bucket-abcdefghij-my-project` for a bucket named "my-bucket" and a project id "my-project" where `abcdefghij` is the 10 character random string, is that ok? if the project id must be prepended, we may have to work with that, but we'd prefer if it would be appended instead. What are your thoughts?